pmm-app/tests/upgradePMM_test.js

```diff
@@ -4,92 +4,132 @@ const serviceNames = {
+  async (I, homePage, inventoryAPI, addInstanceAPI) => {
+    // Adding instances for monitoring
+    for (const type of Object.values(addInstanceAPI.instanceTypes)) {
+      await addInstanceAPI.apiAddInstance(type, serviceNames[type.toLowerCase()]);
     }
+    // Checking that instances are RUNNING
+    for (const service of Object.values(inventoryAPI.services)) {
+      await inventoryAPI.verifyServiceExistsAndHasRunningStatus(service, serviceNames[service.service]);
+    }
+  }
 );
 
 Scenario(
-    'Verify user can see Update widget @pmm-upgrade @visual-test @not-pr-pipeline',
-    async (I, adminPage, homePage) => {
-      I.amOnPage(homePage.url);
-      homePage.verifyPostUpdateWidgetIsPresent();
-    }
+  'Verify user is able to Upgrade PMM version @pmm-upgrade @visual-test @not-pr-pipeline',
+  async (I, inventoryAPI, homePage) => {
+    I.amOnPage(homePage.url);
+    await homePage.upgradePMM();
+  }
 );
 
 Scenario(
-    'Verify user can see News Panel @pmm-upgrade @visual-test @not-pr-pipeline',
-    async (I, adminPage, homePage) => {
-      I.amOnPage(homePage.url);
-      I.waitForVisible(homePage.fields.newsPanelTitleSelector, 30);
-      I.waitForVisible(homePage.fields.newsPanelContentSelector, 30);
-      const newsItems = await I.grabNumberOfVisibleElements(`${homePage.fields.newsPanelContentSelector}/div`);
-      assert.ok(newsItems > 1, 'News Panel is empty');
+  'Verify Agents are RUNNING after Upgrade (API) @pmm-upgrade @visual-test @not-pr-pipeline',
+  async (I, inventoryAPI) => {
+    for (const service of Object.values(inventoryAPI.services)) {
+      await inventoryAPI.verifyServiceExistsAndHasRunningStatus(service, serviceNames[service.service]);
     }
+  }
 );
 
 Scenario(
-    'Verify Agents are RUNNING after Upgrade (UI) @pmm-upgrade @visual-test @not-pr-pipeline',
-    async (I, adminPage, pmmInventoryPage) => {
-      I.amOnPage(pmmInventoryPage.url);
-      for (const service of Object.values(serviceNames)) {
-        await pmmInventoryPage.verifyAgentHasStatusRunning(service);
-      }
+  'Verify user can see Update widget @pmm-upgrade @visual-test @not-pr-pipeline',
+  async (I, adminPage, homePage) => {
+    I.amOnPage(homePage.url);
+    homePage.verifyPostUpdateWidgetIsPresent();
+  }
+);
+
+Scenario(
+  'PMM-T262 Open PMM Settings page and verify DATA_RETENTION value is set to 2 days after upgrade @pmm-upgrade @visual-test @not-pr-pipeline',
+  async (I, pmmSettingsPage) => {
+    const dataRetention = '2';
+    I.amOnPage(pmmSettingsPage.url);
+    pmmSettingsPage.waitForPmmSettingsPageLoaded();
+    const dataRetentionActualValue = await I.grabValueFrom(pmmSettingsPage.fields.dataRetentionCount);
+    assert(
+      dataRetention,
+      dataRetentionActualValue,
+      'The Value for Data Retention is not the same as passed via Docker Environment Variable'
+    );
+  }
+);
+
+Scenario(
```

Comment:
I find it unusual to verify the presence of the news panel in a test called `upgradePMM_test.js`. Shouldn't each test be responsible for only one specific task? In this case one could expect us to only test whether the upgrade is successful or not, pls some post-upgrade sanity checks.

Reply:
yes, this is part of our post-upgrade sanity checks, so we are executing it.